orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -626,6 +731,24 @@ Status TransformGraphForPipeline(
     ResolveForTraining(graph, weights_to_train);
   }
 
+  // If user wants to keep original outputs, we add fake outputs if the
+  // current graph partition doesn't produce them.
+  for (size_t i = 0; keep_original_output_schema && i < expected_output_names.size(); ++i) {
```

Comment:
>keep_original_output_schema [](start = 21, length = 27)

nit: maybe put the loop inside a `if(keep_original_output_schema)` block instead of checking it everything for each loop iteration?

Reply:
Sure.

---
In reply to: [538786101](https://github.com/microsoft/onnxruntime/pull/5815#discussion_r538786101) [](ancestors = 538786101)